### ISSUE TITLE
Content Ratings expansion for Plex OFDB bundle

### ISSUE THREAD

**Issue by rossinior** [sander1/OFDB.bundle#3]
At the moment, you are scanning the ofdb.de movie page for "shop articles" that have the content rating for FSK.

While this is fine for actual products which have DVDs or Blurays out which are available, there are plenty of examples which do have FSK ratings available via ofdb.de, but which are not caught by the agent.

Here are some of many examples of movies that have FSK content rating available, but are not scanned thoroughly enough by OFDB.bundle.

http://www.ofdb.de/film/11309,und-das-Leben-geht-weiter
http://www.ofdb.de/film/11998,36-Stunden
http://www.ofdb.de/film/34352,Am-fernen-Horizont
...
(all of these have no FSK content rating in page, but FSK ratings in all of the German Fassungen)

These movie has "Fassungen", that can be followed via separate links: ...

view.php?page=fassung&fid=<movieid>&vid=<fassungid>

In most of these, the FSK is listed in a table, even if the main movie page does not show any.
As this means loading more web pages, you could have the scan as it is, but expand to load all "Deutsche Fassungen" if you have not found any content rating with the usual search.

I have to say, that not all of the "Fassungen" hold content ratings, these are more common with "Kino", "DVD", "Video", "Blu-ray" Fassungen that with "TV"...

In case you have more than one (different) FSK ratings: those "Fassungen" have play times given, so you could estimate with the move play time, which FSK is probably the right one.

**Comment by sander1** [comment]
Looking at those sub pages for content ratings is a smart idea, I like it! I'll see what I can come up with in code!

**Comment by rossinior** [comment]
Glas you like the ideal. I am looking forward on what you can come up with.

On the matter of Articles being attached to the title, separated Bytes a
coron...
Ich looked at OFDB ruleset, and their policy is to doch that with every
article in any language. That means we need a more sophisticated solution
for that.

What about the following:
If the last word of a title is not longer than 3 characters and separated
by a colon plus space char then put it first.
The only articles longer than 3 chars that need to be handled the same way
are "Eine", "Einer", "Eines".

That would need some testing... ;)

Cheers,
Oliver

Am 13.01.2017 19:46 schrieb "Sander Spies" <notifications@github.com>:

**Comment by sander1** [comment]
The FSK lookup has been updated (https://github.com/plexinc-agents/OFDB.bundle/commit/cd3b6c4b745a9116eff0ce375229a2305891a5b6).

Please consider `plexinc-agents/OFDB.bundle` for future updates and issues (I'm going to delete the `sander1/OFDB.bundle` one).

**Comment by rossinior** [comment]
Hi Sander1,

cool. I will be gald to test it once it got updated on my PMS.

I saw that you have some good skill in programming communication with PMS
(f.e. regarding authentication token et al). Are you interested in Alexa
skill programming?

2017-01-17 16:49 GMT+01:00 Sander Spies <notifications@github.com>:

**Comment by sander1** [comment]
> Are you interested in Alexa skill programming?

Hi! What exactly is Alexa skill programming?

**Comment by rossinior** [comment]
Hi Sander,

I just tested it and it works. You are very talented. Thank you...
I can now unlock many of those Plex fields for content ratings of movies,
that I did manually :))

Alexa skill...

"Amazon Echo" is a voice recognition system (incl. microphone Array and a
loudspeaker) which uses apps (called skills) to provide Solutions to
execute things (f.e. control the lights, Play Music, get sports News to you.

And it would be the damn perfect device to voice control your plex devices.
Since every Plex Server has a web server and is capable of receiving
"commands" via the URL, I think it would be possible to Play playlists,
maybe push movies to other Plex playing devices.

For exmple, it would be so nice to say

"Alexa, tell Plex to play the movie Ten Commandments from Classic Movies
library on my living room tv." ... or ...
"Alexa, tell Plex to play a movie with Denzel Washington from any library
in Sandra's room".

The IP of the Plex Server would be set in a short app (as well as Plex
username and Password).

The "movie/Music/TV Show" set of key words would Trigger a different set of
available commands
The "item title", "library Name" and certain other keywords would be used
to let the Plex Server push a movie to a Plex Playing device known to the
Server.
Maybe one Needs a Rasberry-PC in the local Network running some kind of
service to trigger that if the Server itself cannot do it.


2017-01-18 0:24 GMT+01:00 Sander Spies <notifications@github.com>:
